libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,104 @@
+import { NgControl, NgForm } from '@angular/forms';
+import { Component, ChangeDetectorRef, Self, Optional, Input, ViewChild, AfterViewInit } from '@angular/core';
+import { CheckboxComponent as CoreCheckboxComponent, stateType } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { Status } from './../form-control';
+
+let uniqueId = 0;
+
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** reference of child checkbox implementation */
+    @ViewChild(CoreCheckboxComponent)
+    corecheckbox: CoreCheckboxComponent;
+
+    /** set to true if binary checkbox */
+    @Input()
+    binary: boolean = false;
+
+    /** Sets label for checkbox. */
+    @Input()
+    label: string;
+
+    /** indeterminate state */
+    @Input()
+    tristate: boolean = false;
+
+    /** indeterminate state can be selected  */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** value for checkbox, when it is not binary */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(value: any) {
+        if (value) {
```

Comment:
check parent implementation what it is doing, when setting a value, I need to emit event to parent formField so form field can call markForCheck. 

there is `stateChanges`  - please check it out. 

Also what is the added value set your value here instead of setValue from parent?